### ISSUE TITLE
Feat: Support for Python 3.10

### ISSUE THREAD

**Issue by dgalligani** [macs3-project/MACS#498]
Hello, I've problems installing MACS using the latest python version, 3.10.1  As I am maintaining a production environment, I cannot downgrade the python version to 3.8 (which is 2 years old)

Please include support for this version :)

Many thanks

**Comment by taoliu** [comment]
Fixed in #497 :)

**Comment by millerh1** [comment]
Hi @taoliu just wondering if [this change](https://github.com/macs3-project/MACS/pull/497/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7L46) might contain a typo... I currently can't install this with pip:

```shell
(macs3) millerh1@DESKTOP-D6G0FNB:~/RLPipes$ pip install macs3
Collecting macs3
  Using cached MACS3-3.0.0a6.tar.gz (606 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /home/millerh1/miniconda3/envs/macs3/bin/python /home/millerh1/miniconda3/envs/macs3/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmp3r1znaa8
       cwd: /tmp/pip-install-qz0w8gig/macs3_ba72f544defb481db7bade126eac2a2e
  Complete output (1 lines):
  CRITICAL: Python version must >= 3.6!
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/51/59/055d6b44aa28a2d90af59a676b2c55cd0d097424df7979fe721279cd5193/MACS3-3.0.0a6.tar.gz#sha256=03b3df09f1373cb126d491ac08c1735595540e3e705ba0f3ee7c2a6e8c7024b1 (from https://pypi.org/simple/macs3/) (requires-python:>=3.6). Command errored out with exit status 1: /home/millerh1/miniconda3/envs/macs3/bin/python /home/millerh1/miniconda3/envs/macs3/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmp3r1znaa8 Check the logs for full command output.
  Downloading MACS3-3.0.0a5.tar.gz (588 kB)
     |████████████████████████████████| 588 kB 992 kB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /home/millerh1/miniconda3/envs/macs3/bin/python /home/millerh1/miniconda3/envs/macs3/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmp5h_mqwbj
       cwd: /tmp/pip-install-qz0w8gig/macs3_9445a2e7b0464d5e82586f5a93562e2f
  Complete output (1 lines):
  CRITICAL: Python version must >= 3.6!
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/91/ef/6427bc3023f1ecd3ae9e437ec21031d2b4a18518658a411800cfbf919568/MACS3-3.0.0a5.tar.gz#sha256=eb1b8bd16bac94c98ffadbe458c1919c1b0fa69f73ae0f3600716ac835a35012 (from https://pypi.org/simple/macs3/) (requires-python:>=3.6). Command errored out with exit status 1: /home/millerh1/miniconda3/envs/macs3/bin/python /home/millerh1/miniconda3/envs/macs3/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmp5h_mqwbj Check the logs for full command output.
  Downloading MACS3-3.0.0a4.tar.gz (597 kB)
     |████████████████████████████████| 597 kB 1.8 MB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /home/millerh1/miniconda3/envs/macs3/bin/python /home/millerh1/miniconda3/envs/macs3/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmp9bd_6sae
       cwd: /tmp/pip-install-qz0w8gig/macs3_363e44cba46a48fba9d4a6f9a01227ca
  Complete output (1 lines):
  CRITICAL: Python version must >= 3.6!
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/be/30/5e480a0edc04c9a0ed3442323e3c61f6f1e8d6b379d1f578f164243452f5/MACS3-3.0.0a4.tar.gz#sha256=736c47a9c136213741169e3f77899add6181639ec494831afac534f5d02f9e8c (from https://pypi.org/simple/macs3/) (requires-python:>=3.6). Command errored out with exit status 1: /home/millerh1/miniconda3/envs/macs3/bin/python /home/millerh1/miniconda3/envs/macs3/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmp9bd_6sae Check the logs for full command output.
  Downloading MACS3-3.0.0a3.tar.gz (597 kB)
     |████████████████████████████████| 597 kB 4.3 MB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /home/millerh1/miniconda3/envs/macs3/bin/python /home/millerh1/miniconda3/envs/macs3/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmpj398s88z
       cwd: /tmp/pip-install-qz0w8gig/macs3_d0f68ba2bd5b455ea1eb73a95c277a54
  Complete output (1 lines):
  CRITICAL: Python version must >= 3.6!
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/f5/ad/2fd90c46b59c4ae2b5e7a57bd7d4b71e89e661ec7d446b2a969fa7416038/MACS3-3.0.0a3.tar.gz#sha256=f4b6e14113468eb11e4b7b9b28c6e1f6b163662e488bab3d64a7610b5524303f (from https://pypi.org/simple/macs3/) (requires-python:>=3.6). Command errored out with exit status 1: /home/millerh1/miniconda3/envs/macs3/bin/python /home/millerh1/miniconda3/envs/macs3/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmpj398s88z Check the logs for full command output.
  Downloading MACS3-3.0.0a2.tar.gz (245 kB)
     |████████████████████████████████| 245 kB 7.2 MB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /home/millerh1/miniconda3/envs/macs3/bin/python /home/millerh1/miniconda3/envs/macs3/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmpvafc_z6t
       cwd: /tmp/pip-install-qz0w8gig/macs3_a3eb47c9011d412cbb0cae17548a1276
  Complete output (1 lines):
  CRITICAL: Python version must >= 3.6!
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/5f/36/c760516cdd089df00203bc08e44beb2ad47e485fda0b360ee68502ec42f5/MACS3-3.0.0a2.tar.gz#sha256=d0a8bae07f5a5d72d0d60816350d217b9431a80dfcd29d8abc05540b0bdfce34 (from https://pypi.org/simple/macs3/) (requires-python:>=3.6). Command errored out with exit status 1: /home/millerh1/miniconda3/envs/macs3/bin/python /home/millerh1/miniconda3/envs/macs3/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmpvafc_z6t Check the logs for full command output.
  Downloading MACS3-3.0.0a1.tar.gz (137 kB)
     |████████████████████████████████| 137 kB 8.3 MB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /home/millerh1/miniconda3/envs/macs3/bin/python /home/millerh1/miniconda3/envs/macs3/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmp3ep08oao
       cwd: /tmp/pip-install-qz0w8gig/macs3_a553611193a1475d82bac927308666cf
  Complete output (1 lines):
  CRITICAL: Python version must >= 3.6!
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/36/ff/98453c383475cf799d483fbfb27be0a4d979e21999af3985555c49e9a578/MACS3-3.0.0a1.tar.gz#sha256=e05f981231d8c83b2a98fac9fd3875cb25eb137acb08d5e90b5b8516a57ddc03 (from https://pypi.org/simple/macs3/) (requires-python:>=3.6). Command errored out with exit status 1: /home/millerh1/miniconda3/envs/macs3/bin/python /home/millerh1/miniconda3/envs/macs3/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmp3ep08oao Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement macs3 (from versions: 3.0.0a1, 3.0.0a2, 3.0.0a3, 3.0.0a4, 3.0.0a5, 3.0.0a6)
ERROR: No matching distribution found for macs3

```

**Comment by taoliu** [comment]
@millerh1  I will upload a new release this week with the mentioned patch #497 The change hasn't been included in the current 3.0.0a6 on PyPI.

**Comment by dgalligani** [comment]
Oh, many thanks! 🙂
________________________________

**Comment by taoliu** [comment]
Fixed in the release v3.0.0a7 https://github.com/macs3-project/MACS/releases/tag/v3.0.0a7
